qendpoint-store/src/main/java/com/the_qa_company/qendpoint/model/HDTValue.java

```diff
@@ -13,7 +13,7 @@ static int compare(HDTValue v1, HDTValue v2) {
 			return c;
 		}
 
-		return Long.compare(v1.getHDTPosition(), v2.getHDTPosition());
+		return Long.compare(v1.getHDTId(), v2.getHDTId());
```

Comment:
The HDTPosition is checked above, so I assume that the ID should be compared if the positions are equal.